base/src/test/java/uk/ac/ebi/atlas/utils/GeneSetEnrichmentClientTest.java

```diff
@@ -1,42 +1,166 @@
 
+        assertThat(result.getRight().equals(Optional.empty()));
+        assertThat(result.getLeft().isPresent());
+    }
+
+    @Test
+    public void testEmptyResult() {
+        String url = UriComponentsBuilder.fromUriString(urlPattern).buildAndExpand(ARABIDOPSIS_SPECIES, GENE_IDS).encode().toUriString();
+
+        mockServer.expect(requestTo(url))
+                .andExpect(method(HttpMethod.GET))
+                .andRespond(withSuccess());
+
+        List<String> bioentityIdentifiers = Arrays.asList(GENE_IDS.split("\\W+"));
+        Pair<Optional<Exception>, Optional<JsonArray>> result = subject.fetchEnrichedGenes(ARABIDOPSIS_SPECIES, bioentityIdentifiers);
+
+        assertThat(result.getRight().equals(Optional.empty()));
+        assertThat(result.getLeft().isPresent());
+    }
+
+    @Test
+    public void testMalformedTsvHeaderResult() {
+        String url = UriComponentsBuilder.fromUriString(urlPattern).buildAndExpand(ARABIDOPSIS_SPECIES, GENE_IDS).encode().toUriString();
+
+        mockServer.expect(requestTo(url))
+                .andExpect(method(HttpMethod.GET))
+                .andRespond(withSuccess(TSV_WITH_BAD_HEADER, MediaType.TEXT_PLAIN));
+
+        List<String> bioentityIdentifiers = Arrays.asList(GENE_IDS.split("\\W+"));
+        Pair<Optional<Exception>, Optional<JsonArray>> result = subject.fetchEnrichedGenes(ARABIDOPSIS_SPECIES, bioentityIdentifiers);
+
+        assertThat(result.getRight().equals(Optional.empty()));
+        assertThat(result.getLeft().isPresent());
+    }
+
+    @Test
+    public void testMalformedTsvFileResult() {
+        String url = UriComponentsBuilder.fromUriString(urlPattern).buildAndExpand(ARABIDOPSIS_SPECIES, GENE_IDS).encode().toUriString();
+
+        mockServer.expect(requestTo(url))
+                .andExpect(method(HttpMethod.GET))
+                .andRespond(withSuccess(TSV_WITH_DIFFERENT_LINE_SIZES, MediaType.TEXT_PLAIN));
+
+        List<String> bioentityIdentifiers = Arrays.asList(GENE_IDS.split("\\W+"));
+        Pair<Optional<Exception>, Optional<JsonArray>> result = subject.fetchEnrichedGenes(ARABIDOPSIS_SPECIES, bioentityIdentifiers);
+
+        assertThat(result.getRight().equals(Optional.empty()));
+        assertThat(result.getLeft().isPresent());
+    }
+
+    @Test
+    public void testNaNValuesResult() {
+        String url = UriComponentsBuilder.fromUriString(urlPattern).buildAndExpand(ARABIDOPSIS_SPECIES, GENE_IDS).encode().toUriString();
+
+        mockServer.expect(requestTo(url))
+                .andExpect(method(HttpMethod.GET))
+                .andRespond(withSuccess(TSV_WITH_NAN, MediaType.TEXT_PLAIN));
+
+        List<String> bioentityIdentifiers = Arrays.asList(GENE_IDS.split("\\W+"));
+        Pair<Optional<Exception>, Optional<JsonArray>> result = subject.fetchEnrichedGenes(ARABIDOPSIS_SPECIES, bioentityIdentifiers);
+
+        assertThat(jsonPath("$", hasSize(2)));
+        assertThat(jsonPath("$[0].effect_size", is(Double.NaN)));
```

Comment:
This will produce incorrect JSON:
```
{...,
 "effect_size": NaN,
 ...}
```
I guess a better alternative to avoid a `SyntaxError` from DataTables is for NaN to be a string.